src/test/java/core/basesyntax/BlockingQueueTest.java

```diff
@@ -0,0 +1,74 @@
+package core.basesyntax;
+
+import core.basesyntax.thread.Consumer;
+import core.basesyntax.thread.Producer;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class BlockingQueueTest {
+    private BlockingQueue<Integer> blockingQueue;
+
+    @Test
+    public void put_emptySpaceLeft_ok() throws InterruptedException {
+        blockingQueue = new BlockingQueue<>(100);
+        Thread producer = new Thread(new Producer(blockingQueue));
+
+        producer.start();
+        Thread.sleep(500);
+        Assert.assertNotEquals("Your `put()` method should not put the producer thread "
+                + "in the WAITING state if there is still space in the queue",
+                Thread.State.WAITING, producer.getState());
+    }
+
+    @Test
+    public void put_fullCapacity_ok() throws InterruptedException {
+        blockingQueue = new BlockingQueue<>(49);
```

Comment:
Maybe it's better to call class MyBlockingQueue and avoid confusion with [interface](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/BlockingQueue.html)?
![image](https://user-images.githubusercontent.com/13794896/133201116-06f57894-0d45-484d-af65-94cf7de806b6.png)

Reply:
discussed